### DISCUSSION TITLE
Code differences between opencrvs-core and mosip-mediator

### DISCUSSION THREAD

**Discussion by NagarjunaKuchi** [opencrvs/opencrvs-core#6395] [Technical Q&A]
Hi Team,

We are trying to integrate opencrvs with MOSIP. After subscribing the events, data is not flowing to MOSIP through [mosip-mediator](https://github.com/opencrvs/mosip-mediator). When we analyze the code, found some differences in the code.

opencrvs-core has some [changes](https://github.com/opencrvs/opencrvs-core/commit/ae71174ed226115496a282fcbdaf863769e81676) to support OAuth 2.0 . These changes were not reflecting in [mosip-mediator](https://github.com/opencrvs/mosip-mediator/blob/master/packages/api/src/authToken/opencrvsAuthToken.ts#L33).

We are using **master** branch of [mosip-mediator](https://github.com/opencrvs/mosip-mediator/tree/master) and [opencrvs-core](https://github.com/opencrvs/opencrvs-core/tree/master).

**Code -Differences :**
   [mosip-mediator](https://github.com/opencrvs/mosip-mediator/blob/master/packages/api/src/authToken/opencrvsAuthToken.ts#L33)
[opencrvs-core](https://github.com/opencrvs/opencrvs-core/blob/master/packages/auth/src/server.ts#L375)
[OAuth 2.0 changes ](https://github.com/opencrvs/opencrvs-core/commit/ae71174ed226115496a282fcbdaf863769e81676)  

Please let us know is there any other branch available which is compatible with mosip-mediator.

**Comment by Zangetsu101** [comment]
Hi @NagarjunaKuchi as you pointed out the mosip mediator is a bit behind from the recent changes we made to core. But rest assured as it is in our roadmap towards 1.4.0 to bring it up to date. Although I can see that you already have a rough idea of what might need to be changed, basically the data in the body of request will now need to be provided in the request uri instead. So if you have perhaps already made those changes or you want to do it then a PR would be highly appreciated :smile: .

**Comment by euanmillar** [comment]
Hi @NagarjunaKuchi I just wanted to also point out that we are working with MOSIP to make some changes to mosip-mediator. It is currently coded to be hosted within the MOSIP stack and we are moving it out to be hosted on the OpenCRVS stack instead.  You can expect to see some gradual movement in this repo in the coming weeks.  I will let you know when it is ready for testing again.   As @Zangetsu101 pointed out we will bring it up to date.  FYI our release calendar changed this week.  We are poised to release 1.4 with some other infrastructure changes and this mosip-mediator upgrade will now be in 1.5.  The schedule for 1.5 is currently to be released in beta at the end of March this year I'm afraid.
